tools/download-test.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+while [[ $# -gt 1 ]]
+do
+  key="$1"
+
+  case $key in
+      -t|--tap)
+      TAP="$2"
+      shift # past argument
+      ;;
+      *) # unknown option
+      ;;
+  esac
+  shift # past argument or value
+done
+
+if [ -z ${NODE_VERSION} ]; then
+  echo "NODE_VERSION is undefined! Please declare NODE_VERSION"
+  exit 1
+fi
+# Remove old files
+[ ! -z "${TAP}" ] && rm -r ${TAP} || true
+rm -r v* || true
+
+# Scrape the download server
+rsync -av rsync://unencrypted.nodejs.org/nodejs/release/v${NODE_VERSION} .
+
+num=1
+cat v${NODE_VERSION}/SHASUMS256.txt | { while read line
+do
+   sha=$(echo "${line}" | awk '{print $1}')
+   file=$(echo "${line}" | awk '{print $2}')
+   echo "Checking shasum for $file"
+   remoteSHA=$(shasum -a 256 "nodejs.org/dist/v${NODE_VERSION}/${file}" | awk '{print $1}')
+   if [ ${remoteSHA} != ${sha} ]; then
```

Comment:
@mhdawson is seeing the `unary operator` error because you're not quoting your variables, I suggest dropping the `{}` where they're not needed (everywhere in this script) and putting vars in double quotes everywhere.